R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -3564,11 +3564,14 @@ test_that("gapply() Arrow optimization", {
                      stopifnot(is.numeric(key[[1]]))
                    }
                    stopifnot(class(grouped) == "data.frame")
+                   # mtcars' hp is more then 50.
+                   stopifnot(all(grouped$hp > 50))
```

Comment:
should the test check for # of columns?

Reply:
Yea, that's actually what I did initially. And then, realised that column numbers were actually fine (because correct columns are set to Arrow writers ahead but the problem looks the data itself)

Reply:
Ah, do you mean testing the number of columns while we're here? Yup, we can. I will add them if there are more changes to push. Otherwise, it should be orthogonal.

Reply:
yes, I mean validate the # of col on the R side